metadata-io/src/main/java/com/linkedin/metadata/service/GlossaryTermService.java

```diff
@@ -0,0 +1,469 @@
+  }
+
+  /**
+   * Batch removes multiple glossary terms for a set of resources.
+   *
+   * @param glossaryTermUrns the urns of the terms to remove
+   * @param resources references to the resources to change
+   * @param authentication authentication to use when making the change
+   *
+   */
+  public void batchRemoveGlossaryTerms(
+      @Nonnull List<Urn> glossaryTermUrns,
+      @Nonnull List<ResourceReference> resources,
+      @Nonnull Authentication authentication) {
+    log.debug("Batch adding GlossaryTerms to entities. glossaryTerms: {}, resources: {}", resources, glossaryTermUrns);
+    try {
+      removeGlossaryTermsFromResources(glossaryTermUrns, resources, authentication);
+    } catch (Exception e) {
+      throw new RuntimeException(String.format("Failed to batch add GlossaryTerms %s to resources with urns %s!",
+          glossaryTermUrns,
+          resources.stream().map(ResourceReference::getUrn).collect(Collectors.toList())),
+          e);
+    }
+  }
+
+  private void addGlossaryTermsToResources(
+      List<Urn> glossaryTerms,
+      List<ResourceReference> resources,
+      Authentication authentication
+  ) throws Exception {
+    List<MetadataChangeProposal> changes = buildAddGlossaryTermsProposals(glossaryTerms, resources, authentication);
+    ingestChangeProposals(changes, authentication);
+  }
+
+  private void removeGlossaryTermsFromResources(
+      List<Urn> glossaryTerms,
+      List<ResourceReference> resources,
+      Authentication authentication
+  ) throws Exception {
+    List<MetadataChangeProposal> changes = buildRemoveGlossaryTermsProposals(glossaryTerms, resources, authentication);
+    ingestChangeProposals(changes, authentication);
+  }
+
+  @VisibleForTesting
+  List<MetadataChangeProposal> buildAddGlossaryTermsProposals(
+      List<Urn> glossaryTermUrns,
+      List<ResourceReference> resources,
+      Authentication authentication
+  ) throws URISyntaxException {
+
+    final List<MetadataChangeProposal> changes = new ArrayList<>();
+
+    final List<ResourceReference> entityRefs = resources.stream()
+        .filter(resource -> resource.getSubResource() == null || resource.getSubResource().equals(""))
+        .collect(Collectors.toList());
+    final List<MetadataChangeProposal> entityProposals = buildAddGlossaryTermsToEntityProposals(glossaryTermUrns, entityRefs, authentication);
+
+    final List<ResourceReference> schemaFieldRefs = resources.stream()
+        .filter(resource -> resource.getSubResourceType() != null && resource.getSubResourceType().equals(SubResourceType.DATASET_FIELD))
+        .collect(Collectors.toList());
+    final List<MetadataChangeProposal> schemaFieldProposals = buildAddGlossaryTermsToSubResourceProposals(glossaryTermUrns, schemaFieldRefs, authentication);
+
+    changes.addAll(entityProposals);
+    changes.addAll(schemaFieldProposals);
+
+    return changes;
+  }
+
+  @VisibleForTesting
+  List<MetadataChangeProposal> buildRemoveGlossaryTermsProposals(
```

Comment:
`buildRemoveGlossaryTermsMetadataChangeProposals`? Since `proposal` is an overloaded word

Reply:
oh interesting....

Reply:
i think in the context here... it should be okay. what im thinking for saas is that we'll build a ProposalService or ActionRequestService most likely to complement